analyses/molecular-subtyping-MB/00-mb-select-pathology-dx.Rmd

```diff
@@ -0,0 +1,138 @@
+- [Pathology diagnosis strings for inclusion](#pathology-diagnosis-strings-for-inclusion)
+- [Any `Other` samples that should be included based on `pathology_free_text_diagnosis`](#any-other-samples-that-should-be-included-based-on-pathology_free_text_diagnosis)
+- [Subset to Medulloblastoma based on `pathology_diagnosis`](#subset-to-medulloblastoma-based-on-pathology_diagnosis)
+- [Session Info](#session-info)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+
+## Set up
+
+```{r}
+library(tidyverse)
+```
+
+This is tied to a specific release. 
+
+```{r}
+data_dir <- file.path("..", "..", "data", "release-v17-20200908")
+histologies_file <- file.path(data_dir, "pbta-histologies.tsv")
+```
+
+We're going to save the pathology diagnosis information we'll use to generate the subset files in a directory `input`.
+
+```{r}
+output_dir <- "input"
+output_file <- file.path(output_dir, 
+                         "mb_subtyping_path_dx_strings.json")
+```
+
+## Read in data
+
+```{r}
+histologies_df <- read_tsv(histologies_file)
+```
+
+## Explore the pathology diagnoses
+
+How do `short_histology`, `integrated_diagnosis`, and `pathology_diagnosis` line up for `Medulloblastoma` biospecimens? 
+
+```{r}
+histologies_df %>% 
+  filter(short_histology == "Medulloblastoma" | integrated_diagnosis == "Medulloblastoma" | pathology_diagnosis == "Medulloblastoma") %>%
+  group_by(short_histology, integrated_diagnosis, pathology_diagnosis, experimental_strategy) %>%
+  tally()
+```
+
+Looks like there are two samples that will be added in by using `pathology_diagnosis`. 
+
+Let's take a look at the free text field.
+
+```{r}
+histologies_df %>% 
+  filter(short_histology == "Medulloblastoma") %>%
+  group_by(pathology_free_text_diagnosis) %>%
+  tally() %>%
+  arrange(desc(n))
+```
+
+This is not as uniform, but that's to be expected. 
+There are a couple `pnet` samples in here according to this field, but we will rely on the other field's labels instead. 
+
+## Pathology diagnosis strings for inclusion
+
+Ignoring the `pnet` samples, in general we can include any samples that have `medulloblastoma` in the `pathology_free_text_diagnosis` field. 
+We can add more terms to this later if needed. 
+
+```{r}
+free_text_dx_terms <- c(
+  "medulloblastoma"
+)
+```
+
+## Any `Other` samples that should be included based on `pathology_free_text_diagnosis`
+
+```{r}
+other_mb_biospecimens <- histologies_df %>% 
+  filter(short_histology == "Other", 
+         grepl(paste0(free_text_dx_terms, collapse = "|"), pathology_free_text_diagnosis)) %>% 
+  pull(Kids_First_Biospecimen_ID)
+
+other_mb_biospecimens
+```
+
+There's no samples in `Other` that need to be added at this time, but if they do later, we can add them to the filter statements in the next section. 
+
+### Save the strings we'll use downstream
+
+Create a list with the strings we'll use for inclusion.
+
+```{r}
+terms_list <- list(include_path_dx = "medulloblastoma",
```

Comment:
```suggestion
terms_list <- list(include_path_dx = "Medulloblastoma",
```
For exact matching reasons - you'd then use `%in%` when you implement matching downstream for future-proofing reasons.